node/bft/src/primary.rs

```diff
@@ -569,7 +569,16 @@ impl<N: Network> Primary<N> {
         }
 
         // If the peer is ahead, use the batch header to sync up to the peer.
-        let transmissions = self.sync_with_batch_header_from_peer(peer_ip, &batch_header).await?;
+        let mut transmissions = self.sync_with_batch_header_from_peer(peer_ip, &batch_header).await?;
+
+        // Check that the transmission ids match and are not fee transactions.
+        for (transmission_id, transmission) in transmissions.iter_mut() {
```

Comment:
@ljedrz can you profile this and determine if an optimization is possible with rayon or tokio?

Reply:
Feel free to open a new PR if so

Reply:
Note that the number of transmissions here can be up to 50.